src/node_perf.h

```diff
@@ -171,13 +172,14 @@ class ELDHistogram : public HandleWrap, public Histogram {
   SET_SELF_SIZE(ELDHistogram)
 
  private:
-  static void DelayIntervalCallback(uv_timer_t* req);
+  static void PrepareCallback(uv_prepare_t* handle);
+  static void CheckCallback(uv_check_t* handle);
 
   bool enabled_ = false;
-  int32_t resolution_ = 0;
   int64_t exceeds_ = 0;
   uint64_t prev_ = 0;
-  uv_timer_t timer_;
+  uv_prepare_t prepare_;
+  uv_check_t check_;
```

Comment:
Thinking out loud here … we basically already do this for the CPU profiler from `Environment::StartProfilerIdleNotifier()`. Is it worth merging those, e.g. always enabling the idle notifier + adding the ability to callbacks to the Environment that are called when idle state is entered/left?

Reply:
I agree that this would work. However I feel like it is not worth merging those because it would add quite a lot of complexity and it would also intertwine two independent code files.